File: javascript/crypto-square/crypto-square.js
var CryptoSquare = function (message) {
  this.originalMessage = message;
  this.normalizedMessage = message.toLowerCase().split('').filter(function (c) {
    return c != ' ' && ((c >= '0' && c <= '9') || (c >= 'a' && c <= 'z'));
  });

  this.normalizePlaintext = function () {
    return this.normalizedMessage.join('');
  };

  this.size = function () {
    var square = Math.sqrt(this.normalizePlaintext().length);
    return Math.floor(square % 1 == 0 ? square : square + 1);
  };

  this.plaintextSegments = function () {
    var segments = [];
    var segment = "";
    var count = 0;
    for (var i = 0; i < this.normalizePlaintext().length; i++) {
      segment += this.normalizePlaintext()[i];
      count += 1;
      if (count == this.size()) {
        segments.push(segment);
        segment = "";
        count = 0;
      }
    }
    if (count > 0) {
      segments.push(segment);
    }
    return segments;
  };

  this.ciphertext = function () {
    var cipher = "";
    var segments = this.plaintextSegments();
    for (var i = 0; i < this.size(); i++) {
      for (var j = 0; j < segments.length; j++) {
        if (i < segments[j].length) {
          cipher += segments[j][i];
        }
      }
    }
    return cipher;
  }
};

module.exports = CryptoSquare;

File: csharp/crypto-square/CryptoSquare.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class Crypto
{
  public Crypto(string message)
  {
    var charArray = message.ToCharArray();
    NormalizePlaintext = new string(charArray.Where(c => char.IsLetterOrDigit(c)).ToArray());
    NormalizePlaintext = NormalizePlaintext.ToLower();
  }

  public string NormalizePlaintext { get; }
  public int Size
  {
    get
    {
      var square = Math.Sqrt(NormalizePlaintext.Length);
      return square % 1 == 0 ? (int)square : (int)square + 1;
    }
  }

  public string[] PlaintextSegments()
  {
    var segments = new List<string>();
    var segment = "";
    var count = 0;
    for (var i = 0; i < NormalizePlaintext.Length; i++)
    {
      segment += NormalizePlaintext[i];
      count++;
      if (count == Size)
      {
        segments.Add(segment);
        segment = "";
        count = 0;
      }
    }
    if (count > 0)
    {
      segments.Add(segment);
      segment = "";
      count = 0;
    }
    return segments.ToArray();
  }

  public string Ciphertext()
  {
    var ciperText = new StringBuilder();
    var segments = PlaintextSegments();
    for (var i = 0; i < Size; i++)
    {
      for (var j = 0; j < segments.Length; j++)
      {
        if (i < segments[j].Length)
          ciperText.Append(segments[j][i]);
      }
    }
    return ciperText.ToString();
  }

  public string NormalizeCiphertext()
  {
    var ciperSegments = new List<string>();
    var segments = PlaintextSegments();
    for (var i = 0; i < Size; i++)
    {
      var ciperSegment = "";
      for (var j = 0; j < segments.Length; j++)
      {
        if (i < segments[j].Length)
          ciperSegment += segments[j][i];
      }
      ciperSegments.Add(ciperSegment);
    }
    return string.Join(" ", ciperSegments.ToArray());
  }
}